File: src/highgui.jl
imread(filename::AbstractString) = _Mat(ccall( (:imread, cv2_lib), Ptr{Void},(Ptr{UInt8},), bytestring(filename)))

imshow(winName::AbstractString, mat::Mat) = ccall( (:imshow, cv2_lib), Void, (Ptr{UInt8}, Ptr{Void}), bytestring(winName), mat.handle)

imclose(winName::AbstractString) = ccall( (:imclose, cv2_lib), Void, (Ptr{UInt8},), bytestring(winName))

waitKey(delay::Int) = ccall( (:waitKey, cv2_lib), Void, (Int32,), delay)

namedWindow(winname::AbstractString, flags::Int64) = ccall((:namedWindow, cv2_lib), Void, (Ptr{UInt8}, Cint, ), bytestring(winname), flags)

File: src/cpp/highgui.cpp
#include "highgui.h"
#include <iostream>

using namespace std;

cv::Mat* imread(char *filename)
{
  cv::Mat img = cv::imread(filename);
  cv::Mat *image = new cv::Mat();
  img.copyTo(*image);

  return image;
}

void imshow(char *winName, cv::Mat* mat)
{
  if (!mat->empty())
    {
      cv::imshow(winName, *mat);
    }
  else
    {
      cout << "Empty image" << endl;
    }
}

void waitKey(int delay)
{
  cv::waitKey(delay);
}

void namedWindow(char *winname, int flags)
{
  cv::namedWindow(winname);
}

void imclose(char *winName)
{
  cv::destroyWindow(winName);
}

